google/cloud/security/inventory/inventory_loader.py

```diff
@@ -185,8 +185,43 @@ def _run_pipelines(pipelines):
                          exc_info=True)
             pipeline.status = 'FAILURE'
         run_statuses.append(pipeline.status == 'SUCCESS')
+    _postprocess_statuses(pipelines, run_statuses)
     return run_statuses
 
+def _postprocess_statuses(pipelines, run_statuses):
+    """Look at related `pipelines` and adjust run statuses as needed.
+    Example: if the `groups` pipeline failed then the `group_members` pipeline
+    should be marked as a failure as well.
+
+    Please note: this function will modify the data passed to it via the
+    arguments below if/as needed.
+
+    Args:
+        pipelines (list): List of pipelines that were run.
+        run_statuses (list): a list of booleans indicating whether each
+        pipeline completed successfully or not.
+
+    Returns:
+        bool: `True` if adjustments were made and `False` otherwise.
+    """
+    result = False
+    indices = dict(
+        (v, i) for (i, v) in enumerate(p.RESOURCE_NAME for p in pipelines))
+    grps_idx = indices.get('groups')
+    grp_members_idx = indices.get('group_members')
+    # Do nothing unless the status for both 'group_members' and 'groups' is
+    # present and the latter is `False`.
+    if (grps_idx is None or grp_members_idx is None or run_statuses[grps_idx]):
+        return result
+
+    if run_statuses[grp_members_idx]:
+        result = True
+        run_statuses[grp_members_idx] = False
+        pipelines[grp_members_idx].status = 'FAILURE'
+
+    return result
```

Comment:
The modified ```pipelines``` might need to be returned, so that it can be used in the outer execution for email template.

for example:
```
    return pipelines, run_statuses
````
Then, your caller can be:
```
pipelines, run_statuses = _postprocess_statuses(pipelines, run_statuses)
```

Reply:
There's no need to return, the actual data passed to the function is modified (using python's call-by-object mechanism)

Reply:
Yah, then just remove this return.